Index: mpi4py/docs/source/usrman/conf.py
===================================================================
--- mpi4py.orig/docs/source/usrman/conf.py	2021-11-20 13:24:00.908610477 +0100
+++ mpi4py/docs/source/usrman/conf.py	2021-12-05 21:12:58.548612091 +0100
@@ -245,10 +245,11 @@
 if html_theme == 'default':
     html_copy_source = False
 
-if html_theme == 'sphinx_rtd_theme':
-    html_theme_options = {
-        'analytics_id': 'UA-48837848-1',
-    }
+# don't enable googletagmanager in order to maintain privacy
+#if html_theme == 'sphinx_rtd_theme':
+#    html_theme_options = {
+#        'analytics_id': 'UA-48837848-1',
+#    }
 
 # Add any paths that contain custom static files (such as style sheets) here,
 # relative to this directory. They are copied after the builtin static files,
