Index: mpi4py/test/test_pack.py
===================================================================
--- mpi4py.orig/test/test_pack.py	2024-08-30 11:41:07.974815861 +0200
+++ mpi4py/test/test_pack.py	2024-08-30 11:41:32.395011906 +0200
@@ -3,6 +3,7 @@
 import arrayimpl
 import os, sys, platform
 
+import platform
 
 def allclose(a, b, rtol=1.e-5, atol=1.e-8):
     try:
@@ -88,6 +89,7 @@
                         real_size = pack_size - overhead
                         self.assertGreaterEqual(real_size, 0)
 
+    @unittest.skipMPI("openmpi(<5.0)", platform.machine() == 'sparc64')
     def testPackUnpackExternal(self):
         for array, typecode1 in arrayimpl.loop():
             with arrayimpl.test(self):
