Index: mpi4py/test/test_spawn.py
===================================================================
--- mpi4py.orig/test/test_spawn.py	2021-08-19 23:29:58.575714523 +0200
+++ mpi4py/test/test_spawn.py	2021-08-19 23:44:16.703250131 +0200
@@ -67,7 +67,7 @@
 
 
 @unittest.skipMPI('MPI(<2.0)')
-@unittest.skipMPI('openmpi(<3.0.0)')
+@unittest.skipMPI('openmpi(<5.0.0)')
 @unittest.skipMPI('openmpi(==4.0.0)')
 @unittest.skipMPI('openmpi(==4.0.1)', sys.platform=='darwin')
 @unittest.skipMPI('openmpi(==4.0.2)', sys.platform=='darwin')
